relay-server/src/endpoints/common.rs

```diff
@@ -409,6 +393,37 @@ pub async fn handle_envelope(
     }
 }
 
+fn emit_envelope_metrics(envelope: &Envelope) {
+    let client_name = envelope.meta().client_name();
+    let mut has_transaction = false;
+    let mut has_attachment = false;
+    for item in envelope.items() {
+        has_transaction |= item.ty() == &ItemType::Transaction;
+        has_attachment |= item.ty() == &ItemType::Attachment;
+        metric!(
+            histogram(RelayHistograms::EnvelopeItemSize) = item.payload().len() as u64,
+            item_type = item.ty().name()
+        );
+        metric!(
+            counter(RelayCounters::EnvelopeItems) += 1,
+            item_type = item.ty().name(),
+            sdk = client_name.name(),
+        );
+        metric!(
+            counter(RelayCounters::EnvelopeItemBytes) += item.payload().len() as u64,
+            item_type = item.ty().name(),
+            sdk = client_name.name(),
+        );
+    }
+
+    if has_transaction && has_attachment {
```

Comment:
We do not need to measure baseline here (total number of transactions), because we get that number from `RelayCounters::EnvelopeItems` (ignoring that envelopes can in theory contain multiple transactions).